### DISCUSSION TITLE
Export from ProseMirror

### DISCUSSION THREAD

**Discussion by rowanc1** [dolanmiu/docx#1194] [Show and tell]
Hi all, I wanted to show a package that we are developing at [Curvenote](https://curvenote.com/for/writing) that allows us to export from a ProseMirror document directly to a Microsoft Word, docx file: [prosemirror-docx](https://github.com/curvenote/prosemirror-docx). It is early, but functional and would love help or suggestions on how to improve the library (based on the newly release v7 of docx 🎉), especially from anyone who might be using ProseMirror.

[Prosemirror](https://prosemirror.net) is a really solid online WYSIWYG editor. For our use-case of technical documents at [Curvenote](https://curvenote.com), we also require word/pdf exports, and chose to integrate with `docx`. We will be continuing with our support for cross-references and citations soon, and would love any pointers from folks who have already done some of that!

Let me know if this is helpful, or what changes you would like to see in the library. Looking forward to any feedback.

**Comment by dolanmiu** [comment]
There are a lot of issues which need handling in the issues section if you want to look to improve it!

Another way is to create a validation step in GitHub Actions, so that during e2e, we can be sure to have valid documents

**Reply by dolanmiu** [reply]
Essentially integrate this as part of the tests:

https://github.com/devoidfury/docx-validator

**Reply by rowanc1** [reply]
I will absolutely continue to contribute here! Got a few improvements and docs updates into docx last release. :)

**Reply by dolanmiu** [reply]
Thanks! much appreciated if you do!

**Reply by dolanmiu** [reply]
Is it ok if I add ProseMirror to the README?

**Reply by rowanc1** [reply]
The readme of `docx`? Certainly would love a link back to this work -- I am not one of the core maintainers or ProseMirror (just an enthusiastic advocate!) -- they have quite an active community though, so hopefully this bridge can help join up these two worlds.
